infra/main.bicep

```diff
@@ -0,0 +1,746 @@
+
+@description('Is the index prechunked')
+param azureSearchIndexIsPrechunked string = 'false'
+
+@description('Top K results')
+param azureSearchTopK string = '5'
+
+@description('Enable in domain')
+param azureSearchEnableInDomain string = 'false'
+
+@description('Content columns')
+param azureSearchContentColumns string = 'content'
+
+@description('Filename column')
+param azureSearchFilenameColumn string = 'filename'
+
+@description('Title column')
+param azureSearchTitleColumn string = 'title'
+
+@description('Url column')
+param azureSearchUrlColumn string = 'url'
+
+@description('Name of Azure OpenAI Resource')
+param azureOpenAIResourceName string = '${environmentName}-openai-${resourceToken}'
+
+@description('Name of Azure OpenAI Resource SKU')
+param azureOpenAISkuName string = 'S0'
+
+@description('Azure OpenAI Model Deployment Name')
+param azureOpenAIModel string = 'gpt-35-turbo'
+
+@description('Azure OpenAI Model Name')
+param azureOpenAIModelName string = 'gpt-35-turbo'
+
+param azureOpenAIModelVersion string = '0613'
+
+@description('Orchestration strategy: openai_function or langchain str. If you use a old version of turbo (0301), plese select langchain')
+@allowed([
+  'openai_function'
+  'langchain'
+])
+param orchestrationStrategy string = 'langchain'
+
+@description('Azure OpenAI Temperature')
+param azureOpenAITemperature string = '0'
+
+@description('Azure OpenAI Top P')
+param azureOpenAITopP string = '1'
+
+@description('Azure OpenAI Max Tokens')
+param azureOpenAIMaxTokens string = '1000'
+
+@description('Azure OpenAI Stop Sequence')
+param azureOpenAIStopSequence string = '\n'
+
+@description('Azure OpenAI System Message')
+param azureOpenAISystemMessage string = 'You are an AI assistant that helps people find information.'
+
+@description('Azure OpenAI Api Version')
+param azureOpenAIApiVersion string = '2023-07-01-preview'
+
+@description('Whether or not to stream responses from Azure OpenAI')
+param azureOpenAIStream string = 'true'
+
+@description('Azure OpenAI Embedding Model Deployment Name')
+param azureOpenAIEmbeddingModel string = 'text-embedding-ada-002'
+
+@description('Azure OpenAI Embedding Model Name')
+param azureOpenAIEmbeddingModelName string = 'text-embedding-ada-002'
+
+@description('Azure Cognitive Search Resource')
+param azureCognitiveSearchName string = '${environmentName}-search-${resourceToken}'
+
+@description('The SKU of the search service you want to create. E.g. free or standard')
+@allowed([
+  'free'
+  'basic'
+  'standard'
+  'standard2'
+  'standard3'
+])
+param azureCognitiveSearchSku string = 'standard'
+
+@description('Azure Cognitive Search Index')
+param azureSearchIndex string = '${environmentName}-index-${resourceToken}'
+
+@description('Azure Cognitive Search Conversation Log Index')
+param azureSearchConversationLogIndex string = 'conversations'
+
+@description('Name of Storage Account')
+param storageAccountName string = '${environmentName}str'
```

Comment:
Shouldn't this have a ${resourceToken} otherwise it resolves to `devstr` which won't be a unique storage account name.

Reply:
I changed this to
`${environmentName}str${resourceToken}` and my worked (but my environment is simply dev)

Reply:
Modified according to @ross-p-smith suggestion, it is worth mentioning that `${resourceToken}` is usually a random string of 13 characters in length and the storage name has a maximum limit of 24 characters. This means that the `${environmentName}` need to be simplified a bit. :)